backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/InlineClasses.kt

```diff
@@ -267,17 +292,24 @@ private object IrTypeInlineClassesSupport : InlineClassesSupport<IrClass, IrType
         }
     }
 
-    override fun getFqName(clazz: IrClass): FqNameUnsafe = clazz.descriptor.fqNameUnsafe
     override fun hasInlineModifier(clazz: IrClass): Boolean = clazz.descriptor.isInline
 
     override fun getNativePointedSuperclass(clazz: IrClass): IrClass? {
         var superClass: IrClass? = clazz
-        while (superClass != null && superClass.descriptor.fqNameUnsafe != InteropFqNames.nativePointed)
+        while (superClass != null && (!superClass.symbol.isPublicApi || InteropIdSignatures.nativePointed != superClass.symbol.signature))
             superClass = superClass.getSuperClassNotAny()
         return superClass
     }
 
     override fun getInlinedClassUnderlyingType(clazz: IrClass): IrType =
             clazz.constructors.firstOrNull { it.isPrimary }?.valueParameters?.single()?.type
                     ?: clazz.declarations.filterIsInstance<IrProperty>().single { it.backingField != null }.backingField!!.type
+
+    override fun getPackageFqName(clazz: IrClass) =
+            clazz.packageFqName
+
+    override fun getName(clazz: IrClass): Name? =
+            clazz.name
+
+    override fun isNestedClass(clazz: IrClass): Boolean = clazz.parent !is IrPackageFragment
```

Comment:
This is not exactly correct. For example, it would return `true` for local classes too, and it is not clear if local classes are actually nested.
The more straightforward way would be to make `isTopLevel` function instead.

Reply:
(The semantics is also different from descriptor version, the proposal above should fix this too).